src/main/kotlin/com/getkeepsafe/dexcount/sdkresolver/SdkResolver.kt

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2016 KeepSafe Software
+ * Copyright 2014 Jake Wharton
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.getkeepsafe.dexcount.sdkresolver
+
+import org.gradle.api.Project
+import org.gradle.api.logging.Logging
+import org.gradle.api.tasks.StopExecutionException
+
+import com.android.SdkConstants.ANDROID_HOME_ENV
+import com.android.SdkConstants.FN_LOCAL_PROPERTIES
+import com.android.SdkConstants.PLATFORM_WINDOWS
+import com.android.SdkConstants.SDK_DIR_PROPERTY
+import com.android.SdkConstants.NDK_DIR_PROPERTY
+import com.android.SdkConstants.currentPlatform
+import java.io.BufferedWriter
+import java.io.File
+import java.io.FileOutputStream
+import java.io.OutputStreamWriter
+import java.util.*
+
+/**
+ * This was lifted and modified from Jake Wharton's SDK Manager gradle plugin:
+ * https://github.com/JakeWharton/sdk-manager-plugin
+ */
+class SdkResolver {
```

Comment:
Because of the way our compilers are ordered, Groovy classes aren't visible to Kotlin code.  Seemed simpler to port this class than to figure out the (inherently transient) build-system situation.